### PR TITLE
Use ELF for Dart AOT snapshots on Fuchsia.

### PR DESCRIPTION
**Pull request by sjindel-google**
This removes the last uses of AOT blob snapshots from Flutter.

### CODE REVIEW

File: shell/platform/fuchsia/dart_runner/mapped_resource.cc [hunk 1]
```diff
@@ -97,4 +108,43 @@ MappedResource::~MappedResource() {
   }
 }
 
+bool ElfSnapshot::Load(fdio_ns_t* namespc, const std::string& path) {
+  fuchsia::mem::Buffer vmo;
+  if (!OpenVmo(&vmo, namespc, path, /*executable=*/true))
+    return false;
```
Comment (chinmaygarde) [inline shell/platform/fuchsia/dart_runner/mapped_resource.cc]: Nit: Maybe log here that we could not open the VMO. Would help debug errors.

Comment (sjindel-google) [reply shell/platform/fuchsia/dart_runner/mapped_resource.cc]: Done.

**Comment by cbracken** [comment]
@sjindel-google ping just so this doesn't get forgotten :)

**Comment by sjindel-google** [comment]
@cbracken I'm (still) trying to test this on Fuchsia :)

**Comment by sjindel-google** [comment]
After discussion with Kaushik, it will be easier to test if I make the same change to the Flutter runner as well.

**Comment by sjindel-google** [comment]
Update: Fuchsia's `dlopen()` can't load our generated ELF, but I've ported our custom ELF loader to Fuchsia and got the spinning cube example working.

**Comment by sjindel-google** [comment]
Once https://dart-review.googlesource.com/c/sdk/+/127641 rolls into Flutter, this should be good to go.

I've tested this with the spinning-cube example in the Fuchsia tree.

**Comment by sjindel-google** [comment]
The dependent change has rolled into Flutter, so this should be ready to go.

**Comment by chinmaygarde** [comment]
I can't explain the presubmit error. It is trying to check a file that has moved. @iskakaushik Do you have any ideas?

**Comment by jason-simmons** [comment]
The presubmit script is flagging the formatting of `shell/platform/fuchsia/dart_runner/BUILD.gn`.

Delete the space before the comma on the lines that were added to that file.

Or try running the `ci/check_gn_format.py` script.

**Comment by sjindel-google** [comment]
I have tested this locally, with and without ELF available.

I'm just waiting for the engine build to turn green again to land.

**Comment by zanderso** [comment]
The flutter tool should also be updated:
https://github.com/flutter/flutter/blob/f118b638ed10bf775fb5430f1b0bdd5de8e3717a/packages/flutter_tools/lib/src/fuchsia/fuchsia_build.dart#L104

**Pull request merged**